### ISSUE TITLE
Fetching dashboards from the API

### ISSUE THREAD

**Issue by dunnkers** [RUGSoftEng/2018-PyDash.io#77]
The issue is more of an informative question to the back-end.

Last Monday we have been working on integrating login into the front-end through the API that was built. The `/api/login` endpoint worked out just fine, however, when trying to fetch `/api/dashboards`, the server seemed to be halting, as if there was no method attached to the endpoint.

Are we doing something wrong or aren't those endpoints implemented yet?

**Comment by TheVeggydude** [comment]
It ought to return a bit of json with mock dashboard data. When looking at the code in `development` that is what it should do. Have not run in myself however. You need to be logged in for it to work however. Are you logged in?

**Comment by dunnkers** [comment]
We seemed to be correctly logged in; we tried fetching the dashboard when **not** logged in, and that returned an `UNAUTHORIZED` HTTP code. In an attempt when we were logged in, it did not return an `UNAUTHORIZED` but stalled instead. Again, it might be our fault somewhere, it could also be that you guys missed something somewhere.

**Comment by Qqwy** [comment]
Which branch are you working on? I'll check it out, as soon as I know that info.

**Comment by dunnkers** [comment]
The front-end/dashboard branch :)

On 28 Mar 2018 12:01, "Qqwy" <notifications@github.com> wrote:

Which branch are you working on? I'll check it out, as soon as I know that
info.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/RUGSoftEng/2018-PyDash.io/issues/77#issuecomment-376831524>,
or mute the thread

**Comment by InputUsername** [comment]
Is this fixed?

**Comment by dunnkers** [comment]
@LarsDoorenbos and @i32enrea will find out as soon as they start working on it!

**Comment by Qqwy** [comment]
This is fixed on #103 .
Tl;Dr: CORS is horrible:
- You'll have to resond with two proper headers from the webserver.
- You'll have to override the 'sending cookies' setting from the frontend (`withCredentials: true`)
- You'll have to understand how the front-end library `axios` makes its requests; for a long time we silently sent a field called `withCredentials` to the servers, rather than overriding the setting during login, which of course did not work.
